### ISSUE TITLE
GDAL/OGR version 2.0?

### ISSUE THREAD

**Issue by fsteggink** [geopython/stetl#32]
Since GDAL/OGR version 2.0 is out for a while, is it a good idea to switch to this as the minimum version of Stetl? I haven't looked into it yet, but will soon. I'm just starting this discussion, because the handling of GFS files while reading GML seems to be somewhat buggy in version 1.11. See pull request #31. I'm also not very happy with how layer creation options are working right now (see issue #30), but I've no idea whether improvements have been made regarding this in version 2.0.

One of the minimal criteria to switch to the new version is that it is easily available on the most important platforms, i.e. Linux, Mac and Windows.

See http://www.osgeo.org/node/1591 for more information.

**Comment by justb4** [comment]
While GDAL 2.x  has more facilities, it is at the moment not the default 
on many platforms, at least Linux and Mac (e.g. homebrew), so I would be 
hesitant at the moment. It is a bit like Python 3. On Linux and Mac most 
do intalls via packaging like Ubuntu or Debian-GIS or Homebrew/MacPorts 
on Mac. If possible we could try to support both versions but I cannot 
estimate the work required for that.

**Comment by fsteggink** [comment]
I've got the impression that supporting GDAL 2.0 would be much less work (if at all needed) than supporting Python 3.

I agree that at this moment it is too early. We obviously would need to do some serious testing, in order to see if and when it will be worth. It is my intention to report on the progress in this issue.

**Comment by justb4** [comment]
Yes, it gets more serious: GDAL 2 is the default even for Ubuntu 14.04 (trusty) when using UbuntuGIS: https://launchpad.net/~ubuntugis/+archive/ubuntu/ubuntugis-unstable
I ran into a Stetl install issue (Python-GDAL mismatch) when building Stetl with Docker since [setup.py](../blob/master/setup.py) required GDAL-Python < 2. I removed that restriction and could build/run, but no extensive testing yet. Possibly we should make [setup.py](../blob/master/setup.py) GDAL-version agnostic: set the requirement based on the installed GDAL version. Note that default Ubuntu 16.04 (Xenial) still has GDAL 1.11 as default and so has Homebrew and http://www.kyngchaos.com/software/frameworks, the 2 most used package-repos for GDAL on the Mac. For Ubuntu the preferred way to install Python wrappers for GDAL is via the [python-gdal](http://packages.ubuntu.com/xenial/python/python-gdal) package rather than the PyPi version.

But the above is more related to the GDAL/Python-GDAL install-strategies for each of the 3 major platforms. Haven't even covered Windows...It seems that Stetl can work with both GDAL 1 (1.11 preferably) and 2. Requiring GDAL 2-only is possibly a bit too early, or is there some function in Stetl really requiring GDAL 2?
